packages/lodestar/src/chain/interface.ts

```diff
@@ -56,14 +57,12 @@ export interface IBeaconChain extends ChainEventEmitter {
    */
   getENRForkID(): Promise<ENRForkID>;
 
-  getHeadState(): Promise<BeaconState|null>;
```

Comment:
I think we should create a:
```typescript
export interface ITreeStateContext {
  state: TreeBacked<BeaconState>;
  epochCtx: EpochContext;
}
```
and use TreeBacked beacon state here and across the `lodestar` package, rather than currently at the boundary of `StateContextCache`.

eg:
```typescript
interface IBeaconChain {
  ...
  getHeadStateContext(): Promise<ITreeStateContext | null>;
  getHeadState(): Promise<TreeBacked<BeaconState> | null>;
}
```

We can also use `ITreeStateContext` as the `StateContextCache` cache item.

Reply:
Why does rest of lodestar care about underlying backing? Isn't it only important inside state cache?

Reply:
serving proofs :)
Also, because we rely on the efficiency of a tree-backed state, and practically can't run a production beacon node without it. If we accept that at the `lodestar` level, then there's not really a benefit to obscuring the backing.

This is somewhat different than the at the level of the state transition, in `lodestar-beacon-state-transition`. There, the actual runtime is not known. We don't know where/what the state transition is being used for. We're just exporting a few functions. There, we do get a benefit from obscuring the backing, since it opens options to downstream consumers. (to use whichever backing suits their application)

Reply:
Can we do that in separate PR?

Reply:
Yeah, lets do that in a separate PR (once we merge 0.12.x back to master)